### PR TITLE
Feature/cqi 149: password validator for user form creation

### PR DESCRIPTION
**Pull request by lruzicki**
Ticket:
https://openimis.atlassian.net/browse/CQI-149

Changes:
- password validator is added
- new reducer and actions are added to fetch password policy from backend
- user master panel now validate password before saving (min. medium strength to save password)
- recommendations uses translations (except the library recommendations)

How was it tested?
1. Messages prioritize password policy set in the backend.
2. If password policy is met then library validation enters, checking for additional changes, looking for typical words, password and dates.

Examples of use:
1. Weak password - password policy errors
Screenshot from 2024-05-15 18-01-22

2. Password policy met but the password is still weak
Screenshot from 2024-05-15 18-01-25

3. Medium password, user can save the form, but it recommends him strengthening password (score 2 from library)
Screenshot from 2024-05-15 18-01-34

4. Strong password (score 3 from library)
Screenshot from 2024-05-15 18-01-38

5. Very strong password (score 4 from library)
Screenshot from 2024-05-15 18-01-41

### CODE REVIEW

File: src/components/UserMasterPanel.js
Comment (olewandowski1) [inline src/components/UserMasterPanel.js:30]: Delete comment

File: src/components/UserMasterPanel.js
Comment (olewandowski1) [inline src/components/UserMasterPanel.js:118]: Delete comment

File: src/components/UserMasterPanel.js
Comment (olewandowski1) [inline src/components/UserMasterPanel.js:331]: I suggest you to create a variable which will explicitly say what do you check here. So instead of `color={passwordScore >= 2 ? "primary" : "error"}` I would declare a variable: `const IS_PASSWORD_SECURED = passwordScore >= 2;` and then use it like that: `color={IS_PASSWORD_SECURED ? "primary" : "error"}`. It will be more readable.

**Review by olewandowski1 — CHANGES_REQUESTED** [review]
I left some comments. Please take a look on Sonar analysis, there is one code smell.

File: src/components/UserMasterPanel.js
Comment (lruzicki) [reply src/components/UserMasterPanel.js:30]: ok

File: src/components/UserMasterPanel.js
Comment (lruzicki) [reply src/components/UserMasterPanel.js:118]: ok

**Review by olewandowski1 — APPROVED** [review]
LGTM @lruzicki . I'll merge it once BE will be merged. Let me know about it.

**Comment by lruzicki** [comment]
> LGTM @lruzicki . I'll merge it once BE will be merged. Let me know about it.

BE is merged, you can merge FE.

**Pull request merged**
